Fallout2/Fallout1in2/Mapper/source/scripts/GlobalScripts/gl_tellmeabout.ssl

```diff
@@ -159,14 +140,15 @@ procedure GameModeChange_Handler begin
    end
    else if (tellme_button and (game_mode bwand DIALOG) == false) then begin
       call delete_button;
-      // TODO:
+      // TODO: !!! with that all is well !!!
       // This gets called when using dialog to enter the barter interface.
       // So after using the barter dialog option, TMAs aren't working anymore
       // until the player has restarted the dialog.
       tma_data_array[0] := 0; // Unset TMA data when exiting dialog mode
       tma_gvar_array[0] := 0;
+      debug_msg("Unset TMA data.");
    end
-   else if (tellme_button and (game_mode bwand BARTER or game_mode bwand SPECIAL)) then begin
+   else if (tellme_button and (game_mode bwand SPECIAL or game_mode bwand BARTER)) then begin

```

Comment:
add/replace `game_mode bwand SPECIAL` to `game_mode bwand (DIALOG + SPECIAL) `

Reply:
Ok.

PS: I tried adding in the old dialog_obj check to see if I can fix my bug that way, but it doesn't work anymore either. For some reason, as soon as I click the dialog option to enter barter screen, the game thinks I have exited the dialog and "Unset TMA data." appears in debug log.